sdk/identity/azure-identity/azure/identity/aio/_credentials/chained.py

```diff
@@ -22,6 +24,11 @@ class ChainedTokenCredential(SyncChainedTokenCredential):
     :type credentials: :class:`azure.core.credentials.TokenCredential`
     """
 
+    async def close(self):
```

Comment:
I have a concern here: do we expect customer to "async enter" all the credentials in the chain, or should we have a __aenter__ here that loop thourgh all of them and enter them?
Can I see a sample of usage of this one?

Reply:
This is what I want to enable:
```py
credential = DefaultAzureCredential()
client = FooServiceClient(credential)
# ... time passes, many useful service requests are authorized ...
credential.close()
```
I think `close` is the important API. I don't expect anyone to "enter" or "open" a credential. I have `aenter` doing nothing here because the credential doesn't know which members of its chain will send requests, and transports will open sessions as needed. (At least, our current transport implementations will. Perhaps we should make explicit who's expected to open a transport.)

Reply:
If you implement `__aenter__`/`__aexit__` (e.g. you are an async context manager), then you are strongly signalling to users that using `async with` is general goodness, but you can do the closing yourself if you so see fit.

If we don't want to give an example of intended use with `async with`, then why is it an async context manager?

Reply:
To be consistent with everything else in the SDK that wraps async transport by exposing `__aenter__`, `__aexit__`, and `close`. This PR does add an `async with` example to the README, and it's okay to use credentials that way. Every other credential's `__aenter__` invokes its transport's `__aenter__`.

The awkwardness for `ChainedTokenCredential.__aenter__` is that if it opens sessions for N credentials, N-1 of them may never be used, at some cost dependent on the HTTP client's implementation. These sessions will all be closed by `__aexit__`, but I thought it unnecessary to open them given that our async transports will do so as needed.